### PR TITLE
EIP-1102 updates

### LINKED ISSUE #5941: Lost wakeup with ethereum.enable() using many browser tabs
When a dapp using ethereum.enable() is opened in two browser tabs, reloading two tabs at the same time locks one of them (the other loads successfully).

1. Prepare the following HTML file:
```
<html>
<body>
<script type="text/javascript">
console.log('init');
ethereum.enable().then(() => console.log('enabled'));
</script>
</body>
</html>
```
2. Open the HTML in two browser tabs
3. Refresh the tabs quickly (for me it was not even necessary to use Refresh All)
4. One tab logs 'init' and 'enabled', but the other one logs only 'init'

Both tabs should log 'init' and 'enabled' to the console.

Note that I had granted the permission for MetaMask to inject ethereum account for the location, so no popup appeared during the bug reproduction.
To reproduce, the HTML file location must be http://, file:// does not work.

Software versions:
 - Fedora release 29 (Linux 4.19.6)
 - Chromium 70.0.3538.110
 - MetaMask 5.2.2

### LINKED ISSUE #5951: Background Tab: ethereum enable doesn't wait for response
Possibly linked to this issue: 
https://github.com/MetaMask/metamask-extension/issues/5941

### LINKED ISSUE #5993: [UI] Add a clarification about the "Privacy Mode"
## Problem

Behavior of MetaMask option

> Privacy Mode
> Websites must request access to view your account information.

### Expected Behavior

When OFF, websites do not request access, but gain it automatically.

### Current Behavior

Some Websites always ask the User for access to MM, independent of "Privacy Mode"

## Quick solution

Add a clarification to the UI

"Websites must request access to view your account information. (Note: some websites always request access, even when this option is OFF)

## Some more Info

from: https://github.com/MetaMask/metamask-extension/issues/5971#issuecomment-451137190

> The blog post is accurate: "Even if you haven’t turned Privacy Mode “on” yet, we want to make it clear > what’s happening under the hood, so you may start seeing the “Connect to this site” dialog as you use > your favorite dapps."

Naturally, it is not enough to have this clarification only within a blog-post. It belongs into the UI.

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request includes a few bugfixes around EIP-1102:

- Background <-> dapp messages are parametrized by tab ID
- The privacy notice verbiage was updated to be more accurate
- The "privacy mode" settings option description was updated

Fixes #5941
Fixes #5951
Fixes #5959
Fixes #5993

### CODE REVIEW

**Review by brunobar79 — DISMISSED** [review]
Looks solid to me 💪
(You might have to rerun the tests)

**Comment by rachelhamlin** [comment]
Hey @bitpshr, Status here. What's your target date lately for enabling privacy mode by default? We were planning to do so in our next release (~early Feb).

**Comment by bitpshr** [comment]
@whymarrh @danjm @alextsg Let me know if you have any questions about this PR when you have a chance to review.

File: app/scripts/platforms/extension.js [hunk 1]
```diff
@@ -65,9 +65,11 @@ class ExtensionPlatform {
   }
 
   sendMessage (message, query = {}) {
-    extension.tabs.query(query, tabs => {
+    const id = query.id
+    delete query.id
```
Comment (whymarrh) [inline app/scripts/platforms/extension.js]: Why are we deleting the id from the query here?

File: app/scripts/platforms/extension.js [hunk 2]
```diff
@@ -65,9 +65,11 @@ class ExtensionPlatform {
...
+    extension.tabs.query({ ...query }, tabs => {
       tabs.forEach(tab => {
-        extension.tabs.sendMessage(tab.id, message)
+        extension.tabs.sendMessage(id || tab.id, message)
```
Comment (whymarrh) [inline app/scripts/platforms/extension.js]: If the `id` is passed in and we're querying all the tabs, will this send the same message to the same tab multiple times?

Comment (whymarrh) [reply app/scripts/platforms/extension.js]: Should we be sending a message to tab `id` if that's passed in, else iterate all the tabs?

**Review by whymarrh — COMMENTED** [review]
A few quick comments (also, small nitpick, could we rename `tabID` to `tabId`?)

File: app/scripts/platforms/extension.js (hunk 1 above)
Comment (bitpshr) [reply app/scripts/platforms/extension.js]: Querying tabs worked oddly in Chrome and Firefox.  If the `id` was not deleted before querying, no tabs would ever be returned. This is why this roundabout method of sending to a specific ID is used: if an `id` is present on a `query`, we cache it, delete it, run the rest of the `query`, then only send the resulting message to tabs matching the cached `id`.

File: app/scripts/platforms/extension.js (hunk 2 above)
Comment (bitpshr) [reply app/scripts/platforms/extension.js]: I feel like a query with multiple properties should hit tabs that satisfy _all_ properties, not just some. This is why we 1) query without the ID, then 2) iterate through the results and further filter by ID. If we detected ID first then bailed, that implies an ID takes precedence over all other query properties.

Comment (whymarrh) [reply app/scripts/platforms/extension.js]: Hmm, maybe I don't understand what these APIs are doing well enough. If we call `platform.sendMessage(foo, 42)` we'll be calling `extension.tabs.sendMessage(42, foo)` multiple times (once for each tab returned in the query, no?

Comment (danfinlay) [reply app/scripts/platforms/extension.js]: Whymarrh's concern made sense to me, and then I tried to word it and understood Paul's point:

The [tabs.query method](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/query) does not take an `id` parameter and filter for it, and so this basically allows our own `platform.sendMessage()` method to accept an `id` in its query, and then only send to that tab.

File: ui/app/components/pages/provider-approval/provider-approval.container.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ import { approveProviderRequest, rejectProviderRequest } from '../../../actions'
 
 function mapDispatchToProps (dispatch) {
   return {
-    approveProviderRequest: origin => dispatch(approveProviderRequest(origin)),
-    rejectProviderRequest: origin => dispatch(rejectProviderRequest(origin)),
+    approveProviderRequest: tabID => dispatch(approveProviderRequest(tabID)),
+    rejectProviderRequest: tabID => dispatch(rejectProviderRequest(tabID)),
```
Comment (danfinlay) [inline ui/app/components/pages/provider-approval/provider-approval.container.js]: I would normally raise a concern about mixing platform-specific details (like tab ID) in with the UI, but I have a proposal for a revised approach to this I want to share with you soon (maybe tomorrow?), that I think will help clean up a lot of this tab management logic.

**Review by danfinlay — DISMISSED** [review]
Looks good to me, great to close so many issues with one PR!

**Comment by danfinlay** [comment]
You'll have to pull in latest `develop` to fix (disable) the broken drizzle/beta tests.

**Review by danfinlay — APPROVED** [review]

**Pull request merged**